src/sidebar/components/annotation-action-bar.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+
+const { withServices } = require('../util/service-context');
+const { isShareable, shareURI } = require('../util/annotation-sharing');
+
+const AnnotationActionButton = require('./annotation-action-button');
+const AnnotationShareControl = require('./annotation-share-control');
+
+/**
+ * A collection of `AnnotationActionButton`s in the footer area of an annotation.
+ */
+function AnnotationActionBar({
+  annotation,
+  isPrivate,
+  onDelete,
+  onEdit,
+  onFlag,
+  onReply,
+  groups,
+  permissions,
+  session,
+  settings,
+}) {
+  // Is the current user allowed to take the given `action` on this annotation?
+  const userIsAuthorizedTo = action => {
+    return permissions.permits(
+      annotation.permissions,
+      action,
+      session.state.userid
```

Comment:
`store.profile().userid` is preferred to accessing `session.state`, which is a legacy accessor that was created to support incrementally adapting existing code after the central store was introduced.